tests/integration/pkg/main_test.go

```diff
@@ -0,0 +1,93 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	"sigs.k8s.io/e2e-framework/pkg/env"
+	"sigs.k8s.io/e2e-framework/pkg/envconf"
+	"sigs.k8s.io/e2e-framework/support/kind"
+)
+
+const (
+	envNameUseKubeConfig = "USE_KUBECONFIG"
+	envNameKubeConfig    = "KUBECONFIG"
+)
+
+var (
+	testenv env.Environment
+)
+
+func TestMain(m *testing.M) {
+	if len(os.Getenv(envNameUseKubeConfig)) > 0 {
+		testenv = env.NewWithKubeConfig(os.Getenv(envNameKubeConfig))
+	} else {
+		testenv = env.New()
+		kindClusterName := generateID("cluster")
+
+		// Use pre-defined environment funcs to create a kind cluster prior to test run
+		testenv.Setup(
+			CreateKindCluster(kindClusterName),
+		)
+
+		// Use pre-defined environment funcs to teardown kind cluster after tests
+		testenv.Finish(
+			DestroyKindCluster(kindClusterName),
+		)
+	}
+
+	os.Exit(testenv.Run(m))
+}
+
+type kindContextKey string
+
+func CreateKindCluster(clusterName string) env.Func {
```

Comment:
This is a slightly misleading name, as it actually returns the cluster create function rather than creating one. How about `GetCreateKindClusterFunc`? Same for the destroy one below.

Reply:
`GetCreateKindClusterFunc` would feel to me like a type name rather than an actual `func`/object. The terminology is the same here as in the original https://github.com/kubernetes-sigs/e2e-framework/blob/9169c8740b064b38d4d1533ae0098473cd22fa1d/pkg/envfuncs/kind_funcs.go#L38.

I needed to add it here in order to store the cluster name I believe

Reply:
`GetX` feels like a type name to you? :suspect: Either way, this is a minor thing that I can live with.

Reply:
`GetCreateKindCluster**Func**` feels like a type name